### PR TITLE
Add kn-source-kafka plugin

### PR DESCRIPTION
**Pull request by dsimansk**
Let's see if we can add another one.

/cc @navidshaikh @rhuss

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -1870,13 +1876,16 @@ knative.dev/caching v0.0.0-20190719140829-2032732871ff/go.mod h1:dHXFU6CGlLlbzaW
...
 knative.dev/caching v0.0.0-20200831163411-f6154e7455e2/go.mod h1:jG64HWkBQmFlY8dk1CH884oHtaT/savnqxcZC8vBnvk=
+knative.dev/client v0.17.0/go.mod h1:62s+OqEJ8mBSkP6c3ZmPB3jl1jmOyiKtTPwJUBXJQqY=
 knative.dev/client v0.17.2/go.mod h1:62s+OqEJ8mBSkP6c3ZmPB3jl1jmOyiKtTPwJUBXJQqY=
 knative.dev/eventing v0.17.0 h1:yGJuwN+jWQPCf4XuL44vAjlemLHnbLTBKyhvRBmikpQ=
 knative.dev/eventing v0.17.0/go.mod h1:9NwCSwLnMCKmgz3YQBNax18mSgBjud8CvfsUUVOZ1sA=
 knative.dev/eventing v0.17.5 h1:ORda+gr1YImQB2ArIT5VruF9WQ0Bb8Dw54LDIZrY3xw=
 knative.dev/eventing v0.17.5/go.mod h1:9NwCSwLnMCKmgz3YQBNax18mSgBjud8CvfsUUVOZ1sA=
 knative.dev/eventing-contrib v0.6.1-0.20190723221543-5ce18048c08b/go.mod h1:SnXZgSGgMSMLNFTwTnpaOH7hXDzTFtw0J8OmHflNx3g=
 knative.dev/eventing-contrib v0.11.2/go.mod h1:SnXZgSGgMSMLNFTwTnpaOH7hXDzTFtw0J8OmHflNx3g=
+knative.dev/eventing-contrib v0.14.0 h1:KrDufWSbngkPI7436JqKbd/FPU1XmzZnsy4SWXb5UCY=
```
Comment (dsimansk) [inline go.sum:1887]: :thinking: the plugin is pulling pretty old eventing-contrib version. Should I try to upgrade that one?

File: go.mod [hunk 2]
```diff
@@ -41,4 +42,6 @@ replace (
 	github.com/docker/docker => github.com/docker/docker v1.4.2-0.20200221181110-62bd5a33f707
 	// Darwin cross-build required version
 	golang.org/x/sys => golang.org/x/sys v0.0.0-20200302150141-5c8b2ff67527
+
+	knative.dev/client-contrib/plugins/source-kafka => github.com/dsimansk/kn-source-kafka v0.0.0-20201022153314-e9a483f1b41c
```
Comment (dsimansk) [inline go.mod:46]: That's my fork with slight modification for now. https://github.com/dsimansk/kn-source-kafka/commit/e9a483f1b41c2fcd609726d4bc31d1f2232c710e

**Comment by dsimansk** [comment]
Couple of thoughts from briefly check of plugin's source:

- Kafka's API is still `v1alpha1` - it seems upgradable with minimal change
- Related to the previous eventing-contrib dependency version is `0.14.x`
  - I'm concerned if it still works with only a dependency bump.

File: go.sum (hunk 1 above)
Comment (navidshaikh) [reply go.sum:1887]: +1, we should upgrade that and also see if we trigger the e2e tests for this plugin in midstream
(that would require additional work to setup kafka as well)

File: openshift/e2e-common.sh [hunk 3]
```diff
@@ -303,10 +303,25 @@ install_strimzi() {
...
+kind: KafkaTopic
+metadata:
+  name: test-topic
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  partitions: 100
+  replicas: 1
+EOF
+
```
Comment (navidshaikh) [inline openshift/e2e-common.sh:319]: :+1:

**Comment by dsimansk** [comment]
Ahh, of course I've messed it up.
```
testing: warning: no tests to run
```

File: go.sum (hunk 1 above)
Comment (rhuss) [reply go.sum:1966]: +1 from my side, too

**Comment by rhuss** [comment]
@dsimansk @navidshaikh Where do we stand with this PR ?

**Comment by navidshaikh** [comment]
@rhuss : We'd need some more work on the plugin side, and need to open a fresh one with new repo name / import paths against release-v0.18.4 branch.

**Comment by dsimansk** [comment]
The PR is obsolete now. Closing to avoid any confusion etc..
/close

**Pull request closed without merging**